SIGSEGV in TestInitializeComponentsErrorHandling: os.Setenv race with t.Parallel()

## Bug Report

### Summary

`TestInitializeComponentsErrorHandling` intermittently panics with a **SIGSEGV** (nil pointer dereference) due to a race condition between `os.Setenv` and `t.Parallel()`. The test sets `DATABASE_PATH` to an invalid path, but a concurrent parallel test overwrites it with `:memory:`, causing `initializeComponents` to succeed unexpectedly. The test then calls `err.Error()` on a nil error, triggering the crash.

### CI Run

- **Failed run:** https://github.com/sipico/bunny-api-proxy/actions/runs/21754985731
- **Test:** `TestInitializeComponentsErrorHandling` in `cmd/bunny-api-proxy/main_test.go:1143`
- **Frequency:** Intermittent (race-dependent)

### Stack Trace

```
--- FAIL: TestInitializeComponentsErrorHandling (0.02s)
    main_test.go:1139: initializeComponents should fail with invalid data path
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x120aa32]

goroutine 25 [running]:
testing.tRunner.func1.2({0x132b620, 0x1aaae40})
    /opt/hostedtoolcache/go/1.25.7/x64/src/testing/testing.go:1872 +0x419
testing.tRunner.func1()
    /opt/hostedtoolcache/go/1.25.7/x64/src/testing/testing.go:1875 +0x683
panic({0x132b620?, 0x1aaae40?})
    /opt/hostedtoolcache/go/1.25.7/x64/src/runtime/panic.go:783 +0x132
github.com/sipico/bunny-api-proxy/cmd/bunny-api-proxy.TestInitializeComponentsErrorHandling(0xc000103dc0)
    main_test.go:1143 +0x272
```

### Root Cause Analysis

**Two bugs combine to cause the crash:**

#### Bug 1: `os.Setenv` + `t.Parallel()` race condition

`TestInitializeComponentsErrorHandling` (line 1110) calls both `t.Parallel()` and `os.Setenv("DATABASE_PATH", "/invalid/path/proxy.db")`. Since `os.Setenv` modifies **process-wide** state, any concurrently running test that also sets `DATABASE_PATH` can overwrite the value.

There are **60+ parallel tests** in this file that set `DATABASE_PATH` to `:memory:` via `os.Setenv`. When the scheduler interleaves execution:

1. `TestInitializeComponentsErrorHandling` sets `DATABASE_PATH = "/invalid/path/proxy.db"`
2. Another parallel test sets `DATABASE_PATH = ":memory:"` (overwriting the invalid path)
3. `config.Load()` reads `DATABASE_PATH` as `:memory:` instead of the invalid path
4. `initializeComponents(cfg)` succeeds (returns `nil` error)
5. The test expects an error but gets `nil`

#### Bug 2: Missing early return after nil error check

```go
// main_test.go:1138-1145
_, err = initializeComponents(cfg)
if err == nil {
    t.Error("initializeComponents should fail with invalid data path")  // line 1139 - non-fatal
}

// Verify the error message is informative
if !strings.Contains(err.Error(), "storage initialization failed") {  // line 1143 - CRASH: err is nil
    t.Errorf("error message should be clear, got: %v", err)
}
```

Line 1139 uses `t.Error()` (non-fatal) instead of `t.Fatal()`, so execution continues to line 1143 where `err.Error()` is called on a nil error, causing the SIGSEGV.

### Affected Tests

The same `os.Setenv` + `t.Parallel()` pattern exists in multiple tests. These are all susceptible to similar race conditions:

| Test | Line | Uses `t.Parallel()` | Nil-deref risk |
|------|------|---------------------|----------------|
| `TestInitializeComponentsErrorHandling` | 1110 | Yes | **Yes** (line 1143) |
| `TestInitializeComponentsWithInvalidDataPath` | 302 | No | **Yes** (line 332) |
| `TestRunWithInvalidDatabasePath` | 1029 | No | **Yes** (line 1060) |

All tests that use `os.Setenv` + `t.Parallel()` in this file (60+ tests) are technically unsafe under `-race`, though most won't crash because they set `DATABASE_PATH` to `:memory:` (which always works).

### Suggested Fix

**Immediate fix (prevent SIGSEGV):**
- Change `t.Error` to `t.Fatal` (or add `return`) on lines 1139, 329, 1058 so tests don't fall through to `err.Error()` when `err` is nil

**Proper fix (eliminate race condition):**
- Option A: Remove `t.Parallel()` from all tests that use `os.Setenv` (simplest, but slower test execution)
- Option B: Use `t.Setenv()` (Go 1.17+) which automatically fails if called after `t.Parallel()` — this will surface the incompatibility at test time and force a design decision
- Option C: Refactor `initializeComponents` and `config.Load` to accept config parameters instead of reading from environment, eliminating the need for `os.Setenv` in tests entirely (most robust, largest change)

### Environment

- Go 1.25.7
- Test flags: `-race -coverprofile=coverage.txt -covermode=atomic`
- OS: GitHub Actions runner (Ubuntu)

--- a/cmd/bunny-api-proxy/main_test.go
+++ b/cmd/bunny-api-proxy/main_test.go
@@ -326,7 +326,7 @@ func TestInitializeComponentsWithInvalidDataPath(t *testing.T) {
 
 	_, err = initializeComponents(cfg)
 	if err == nil {
-		t.Error("expected error with invalid data path")
+		t.Fatal("expected error with invalid data path")
 	}
 
 	if !strings.Contains(err.Error(), "storage initialization failed") {
@@ -1026,7 +1026,6 @@ func TestInitializeComponentsLogLevelVariantWorks(t *testing.T) {
 
 // TestRunConfigLoadErrorHandling tests that run() handles config load errors
 func TestRunWithInvalidDatabasePath(t *testing.T) {
-	t.Parallel()
 	oldDatabasePath := os.Getenv("DATABASE_PATH")
 	oldLogLevel := os.Getenv("LOG_LEVEL")
 	oldBunnyAPIKey := os.Getenv("BUNNY_API_KEY")
@@ -1055,7 +1054,7 @@ func TestRunWithInvalidDatabasePath(t *testing.T) {
 
 	err := run()
 	if err == nil {
-		t.Error("run() should return error with invalid database path")
+		t.Fatal("run() should return error with invalid database path")
 	}
 	if !strings.Contains(err.Error(), "storage initialization failed") {
 		t.Errorf("error should mention storage initialization failure, got: %v", err)
@@ -1108,8 +1107,6 @@ func TestInitializeComponentsWithInfoLogLevel(t *testing.T) {
 
 // TestInitializeComponentsErrorDoesNotLeakResources tests that failed initialization cleans up
 func TestInitializeComponentsErrorHandling(t *testing.T) {
-	t.Parallel()
-
 	oldDatabasePath := os.Getenv("DATABASE_PATH")
 	oldLogLevel := os.Getenv("LOG_LEVEL")
 
@@ -1136,7 +1133,7 @@ func TestInitializeComponentsErrorHandling(t *testing.T) {
 
 	_, err = initializeComponents(cfg)
 	if err == nil {
-		t.Error("initializeComponents should fail with invalid data path")
+		t.Fatal("initializeComponents should fail with invalid data path")
 	}
 
 	// Verify the error message is informative

